### PR TITLE
[14.0][MIG] asterisk_click2dial

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

**Comment by cvinh** [comment]
Hello @alexis-via 
I tested your last commit.
It appears that the window from where we click to dial closes by itself and go back to the Discuss page
Do you have this issue too ?
Best regards

**Comment by alexis-via** [comment]
Yes, we need to fix that. Don't hesitate to propose a PR to this branch to fix it.

File: asterisk_click2dial/models/asterisk_server.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        Returns an instance of the Asterisk Manager
+
+        """
+        user = self.env.user
+        ast_server = user.get_asterisk_server_from_user()
+        # We check if the current user has a chan type
+        if not user.asterisk_chan_type:
+            raise UserError(_("No channel type configured for the current user."))
+
+        # We check if the current user has an internal number
```
Comment (Gretel5X) [inline asterisk_click2dial/models/asterisk_server.py]: The comment does not really correspond to the error message, I'd either say `resource` or `internal number`.
Maybe it is clear for pros on Odoo/Asterisk?

File: asterisk_click2dial/models/asterisk_server.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        raise UserError(
+            _(
+                "Connection Test Successfull! Odoo can successfully login to "
+                "the Asterisk Manager Interface."
+            )
+        )
+
+    @api.model
+    def _get_calling_number_from_channel(self, chan, user):
+        """Method designed to be inherited to work with
```
Comment (Gretel5X) [inline asterisk_click2dial/models/asterisk_server.py]: Inconsistent use of `"""`, above the text was after a newline and the closing `"""`as well

File: asterisk_click2dial/models/asterisk_server.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+        finally:
+            ast_manager.Logoff()
+
+        _logger.debug("Calling party number: '%s'", calling_party_number)
+        return calling_party_number
+
+    @api.model
+    def get_record_from_my_channel(self):
+        calling_number = self.env["asterisk.server"]._get_calling_number()
+        # calling_number = "0641981246"
```
Comment (Gretel5X) [inline asterisk_click2dial/models/asterisk_server.py:272]: Is that a valid number for testing? Maybe remove it :)

Comment (Gretel5X) [reply asterisk_click2dial/models/asterisk_server.py:272]: I just realised that that number is there since several years, so nevermind my comment

**Comment by alexis-via** [comment]
I have ported the AGI script to py3, I'll push it tonight.

File: asterisk_click2dial/models/asterisk_server.py (hunk 1 above)
Comment (alexis-via) [reply asterisk_click2dial/models/asterisk_server.py]: Agree, I removed the comment, which doesn't add real value.

File: asterisk_click2dial/models/asterisk_server.py (hunk 2 above)
Comment (alexis-via) [reply asterisk_click2dial/models/asterisk_server.py]: fixed

File: asterisk_click2dial/models/asterisk_server.py (hunk 3 above)
Comment (alexis-via) [reply asterisk_click2dial/models/asterisk_server.py:272]: ok. It just help me locate the place to put my test number.

**Comment by alexis-via** [comment]
Travis is green! If everyone is ok, I'll merge tomorrow.

**Pull request merged**